src/platform/datapath_raw_socket.c

```diff
@@ -204,47 +206,188 @@ CxPlatTryAddSocket(
+            Result =
+                getsockname(
+                    Socket->AuxSocket,
+                    (struct sockaddr*)&TempLocalAddress,
+                    &AssignedLocalAddressLength);
+            if (Result == SOCKET_ERROR) {
+                int WsaError = SocketError();
+                QuicTraceEvent(
+                    DatapathErrorStatus,
+                    "[data][%p] ERROR, %u, %s.",
+                    Socket,
+                    WsaError,
+                    "getsockname");
+                CxPlatRwLockReleaseExclusive(&Pool->Lock);
+                Status = HRESULT_FROM_WIN32(WsaError);
+                goto Error;
+            }
+            LocalPortChosen = TempLocalAddress.Ipv4.sin_port;
+            TempUdpSocket = socket(AF_INET6, SOCK_DGRAM, IPPROTO_UDP);
+            if (TempUdpSocket == INVALID_SOCKET) {
+                int WsaError = SocketError();
+                QuicTraceEvent(
+                    DatapathErrorStatus,
+                    "[data][%p] ERROR, %u, %s.",
+                    Socket,
+                    WsaError,
+                    "temp udp socket");
+                CxPlatRwLockReleaseExclusive(&Pool->Lock);
+                Status = HRESULT_FROM_WIN32(WsaError);
+                goto Error;
+            }
+
+            Option = FALSE;
+            Result =
+                setsockopt(
+                    TempUdpSocket,
+                    IPPROTO_IPV6,
+                    IPV6_V6ONLY,
+                    (char*)&Option,
+                    sizeof(Option));
+            if (Result == SOCKET_ERROR) {
+                int WsaError = SocketError();
+                QuicTraceEvent(
+                    DatapathErrorStatus,
+                    "[data][%p] ERROR, %u, %s.",
+                    Socket,
+                    WsaError,
+                    "Set IPV6_V6ONLY (temp udp socket)");
+                CxPlatRwLockReleaseExclusive(&Pool->Lock);
+                Status = HRESULT_FROM_WIN32(WsaError);
+                goto Error;
+            }
 
+            CxPlatZeroMemory(&TempLocalAddress, sizeof(TempLocalAddress));
+            CxPlatConvertToMappedV6(&Socket->LocalAddress, &TempLocalAddress);
+            TempLocalAddress.Ipv4.sin_port = 0;
+            Result =
+                bind(
+                    TempUdpSocket,
+                    (struct sockaddr*)&TempLocalAddress,
+                    sizeof(TempLocalAddress));
+            if (Result == SOCKET_ERROR) {
+                int WsaError = SocketError();
+                QuicTraceEvent(
+                    DatapathErrorStatus,
+                    "[data][%p] ERROR, %u, %s.",
+                    Socket,
+                    WsaError,
+                    "bind (temp udp socket)");
+                CxPlatRwLockReleaseExclusive(&Pool->Lock);
+                Status = HRESULT_FROM_WIN32(WsaError);
+                goto Error;
+            }
+
+            Result =
+                connect(
```

Comment:
Should we really be calling `connect` on the TCP socket? Doesn't this start the stack's TCP handshake?

Reply:
We are calling `connect` on a temporary **UDP** socket.